src/lib/CurrencyInfo.js

```diff
@@ -0,0 +1,241 @@
+    MKD: 'ден',
+    MMK: 'Ks', // Ks for plural
+    MRU: 'UM',
+    MVR: ['Rf', '.ރ'],
+    MZN: 'MT',
+    NPR: 'रु', // mismatch to Wikipedia as actual wiki article shows it as रु, also in Nepali
+    OMR: ['R.O.', 'ر.ع.'],
+    PAB: 'B/.',
+    PEN: 'S/', // mismatch to Wikipedia as actual wiki article shows it as S/, also in Spanish
+    PKR: '₨',
+    QAR: ['QR', 'ر.ق'],
+    RSD: 'дин.',
+    SAR: ['SR', '﷼'],
+    SDG: ['£SD', 'ج.س.'],
+    SOS: 'Sh.',
+    TJS: 'SM', // mismatch to Wikipedia as actual wiki article shows it as SM
+    TMT: 'm', // mismatch to Wikipedia as actual wiki article shows it as m
+    TND: ['DT', 'د.ت'],
+    UZS: 'сум', // mismatch to Wikipedia as actual wiki article shows it as сум
+    VES: 'Bs.',
+    WST: 'T',
+    XPF: '₣',
+    YER: ['RI', '﷼'],
+};
+
+/**
+ * Some currencies have been devalued so much by inflation that their sub-units have been removed from circulation
+ * or are effectively not being used anymore. This is not for all currencies reflected yet in toLocaleString, such
+ * that we mark some currencies manually as decimal-less.
+ * @private
+ * @readonly
+ * @type {Set<string>}
+ */
+CurrencyInfo.CUSTOM_DECIMAL_LESS_CURRENCIES = new Set([
+    'AMD', // sub-unit rarely used
+    'AOA', // sub-unit rarely used
+    'ARS', // sub-unit discontinued
+    'BDT', // sub-unit discontinued
+    'BTN', // sub-unit rarely used
+    'CDF', // sub-unit rarely used
+    'COP', // sub-unit rarely used
+    'CRC', // sub-unit discontinued
+    'CVE', // sub-unit discontinued
+    'CZK', // sub-unit discontinued
+    'DOP', // sub-unit rarely used
+    'DZD', // sub-unit discontinued
+    'GMD', // sub-unit discontinued
+    'GYD', // sub-unit discontinued
+    'HUF', // sub-unit discontinued
+    'IDR', // sub-unit discontinued
+    'INR', // sub-unit discontinued
+    'JMD', // sub-unit discontinued
+    'KES', // sub-unit rarely used
+    'KGS', // sub-unit rarely used
+    'KHR', // sub-unit discontinued
+    'KZT', // sub-unit rarely used
+    'LKR', // sub-unit discontinued
+    'MAD', // sub-unit rarely used
+    'MKD', // sub-unit discontinued
+    'MNT', // sub-unit discontinued
+    'MOP', // sub-unit discontinued
+    'MWK', // sub-unit rarely used
+    'MXN', // sub-unit rarely used
+    'NGN', // sub-unit rarely used
+    'NOK', // sub-unit discontinued
+    'NPR', // sub-unit rarely used
+    'PHP', // sub-unit rarely used
+    'PKR', // sub-unit discontinued
+    'RUB', // sub-unit rarely used
+    'SEK', // sub-unit discontinued
+    'TWD', // sub-unit discontinued
+    'TZS', // sub-unit discontinued
+    'UAH', // sub-unit discontinued
+    'UYU', // sub-unit discontinued
+    'UZS', // sub-unit discontinued
+    'VES', // sub-unit rarely used
+]);
+
+/**
+ * Cached auto-generated CurrencyInfos such that they do not need to be recalculated.
+ * @private
+ * @readonly
+ * @type {{[currencyAndLocale: string]: CurrencyInfo}}
+ */
+CurrencyInfo.CACHED_AUTO_GENERATED_CURRENCY_INFOS = {};
```

Comment:
Do we really need to cache currency infos? How long does it actually take to calculate?